src/Microsoft.ML.AutoML/API/BinaryClassificationExperiment.cs

```diff
@@ -155,6 +158,161 @@ internal BinaryClassificationExperiment(MLContext context, BinaryExperimentSetti
                   TaskKind.BinaryClassification,
                   TrainerExtensionUtil.GetTrainerNames(settings.Trainers))
         {
+            _experiment = context.Auto().CreateExperiment();
+        }
+
+        public override ExperimentResult<BinaryClassificationMetrics> Execute(IDataView trainData, ColumnInformation columnInformation, IEstimator<ITransformer> preFeaturizer = null, IProgress<RunDetail<BinaryClassificationMetrics>> progressHandler = null)
+        {
+            var label = columnInformation.LabelColumnName;
+            _experiment.SetEvaluateMetric(Settings.OptimizingMetric, label);
+            _experiment.SetTrainingTimeInSeconds(Settings.MaxExperimentTimeInSeconds);
+
+            // Cross val threshold for # of dataset rows --
+            // If dataset has < threshold # of rows, use cross val.
+            // Else, run experiment using train-validate split.
+            const int crossValRowCountThreshold = 15000;
+            var rowCount = DatasetDimensionsUtil.CountRows(trainData, crossValRowCountThreshold);
+            // TODO
+            // split cross validation result according to sample key as well.
+            if (rowCount < crossValRowCountThreshold)
+            {
+                const int numCrossValFolds = 10;
+                _experiment.SetDataset(trainData, numCrossValFolds);
+            }
+            else
+            {
+                var splitData = Context.Data.TrainTestSplit(trainData);
+                _experiment.SetDataset(splitData.TrainSet, splitData.TestSet);
+            }
+
+            MultiModelPipeline pipeline = new MultiModelPipeline();
+            if (preFeaturizer != null)
+            {
+                pipeline = pipeline.Append(preFeaturizer);
+            }
+
+            pipeline = pipeline.Append(Context.Auto().Featurizer(trainData, columnInformation, Features))
+                               .Append(Context.Auto().BinaryClassification(label, Features));
+            _experiment.SetPipeline(pipeline);
+
+            var monitor = new BinaryClassificationTrialResultMonitor();
+            monitor.OnTrialCompleted += (o, e) =>
+            {
+                var detail = ToRunDetail(e);
+                progressHandler?.Report(detail);
+            };
+
+            _experiment.SetMonitor(monitor);
+            _experiment.Run();
+
+            var runDetails = monitor.RunDetails.Select(e => ToRunDetail(e));
+            var bestRun = ToRunDetail(monitor.BestRun);
+            var result = new ExperimentResult<BinaryClassificationMetrics>(runDetails, bestRun);
+
+            return result;
+        }
+
+        public override ExperimentResult<BinaryClassificationMetrics> Execute(IDataView trainData, IDataView validationData, ColumnInformation columnInformation, IEstimator<ITransformer> preFeaturizer = null, IProgress<RunDetail<BinaryClassificationMetrics>> progressHandler = null)
```

Comment:
Couldn't you combine this with the top method and just have the `validationData` default to null?

Reply:
This would change the existing API, which I would rather avoid in this PR. We can simplify these `Execute` methods in a separate PR though

Reply:
Makes sense to me.